### PR TITLE
mongodb cache plugin

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
This PR adds a mongodb fact cache plugin

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
lib/ansible/plugins/cache/mongodb.py

##### ANSIBLE VERSION

```
devel
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/plugins/cache/mongodb.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    """
+    def __init__(self, *args, **kwargs):
+        self._timeout = int(C.CACHE_PLUGIN_TIMEOUT)
+        self._prefix = C.CACHE_PLUGIN_PREFIX
+        self._cache = {}
+        self._managed_indexes = False
+
+    def _manage_indexes(self, collection):
+        '''
+        This funciton manages indexes on the mongo collection.
```
Comment (alikins) [inline lib/ansible/plugins/cache/mongodb.py]: doc typo, s/funcition/function

**Comment by alikins** [comment]
I don't know anything about mongo, but plugin looks good to me.

File: lib/ansible/plugins/cache/mongodb.py (hunk 1 above)
Comment (sivel) [reply lib/ansible/plugins/cache/mongodb.py]: Fixed

**Comment by sivel** [comment]
As a note I canceled the CI build on the last 2 commits, since there was a long queue, and those changes were tangential to the functionality of the plugin.

**Pull request merged**